inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/annotation/DynamicWorkflowDocumentNavigationActionBarExtension.java

```diff
@@ -91,53 +105,106 @@ public Panel createActionBarItem(String aId, AnnotationPageBase aPage)
         return new DynamicDocumentNavigator(aId);
     }
 
-    //Init of the page, select a random document
+    // Init of the page, select a random document
     @Override
     public void onInitialize(AnnotationPageBase aPage)
     {
         User user = aPage.getModelObject().getUser();
         Project project = aPage.getModelObject().getProject();
-        //Check if there is a document in progress and return this one
-        for (AnnotationDocument annotationDocument:
-            documentService.listAnnotationDocuments(project,user)) {
-            //There was one in progress, load it
+        annotationDocumentList = documentService.listAnnotationDocuments(project, user);
```

Comment:
This is slow. Better implement a method in the document service which already filters by state at the SQL level.